internal/slicer/report.go

```diff
@@ -0,0 +1,47 @@
+package slicer
+
+import (
+	"io/fs"
+
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+)
+
+type ReportEntry struct {
+	Path   string
+	Mode   fs.FileMode
+	Hash   string
+	Size   int
+	Slices []*setup.Slice
+	Link   string
+}
+
+// Report holds the information about files and directories created when slicing
+// packages.
+type Report struct {
+	Root string
+	// map indexed by path.
+	Entries map[string]ReportEntry
+}
+
+func NewReport(root string) *Report {
+	return &Report{Entries: make(map[string]ReportEntry), Root: root}
+}
+
+func (r *Report) AddEntry(slice *setup.Slice, entry fsutil.Info) {
+	if info, ok := r.Entries[entry.Path]; ok {
```

Comment:
So the _info_ is named _entry_, and the _entry_ is named _info_?  Beautiful. ;-)

Reply:
You are right, how did I not see that :D